torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -71,8 +71,8 @@ class ProcessGroup {
       const AllreduceOptions& opts = AllreduceOptions()) = 0;
 
  protected:
-  const int rank_;
-  const int size_;
+  int rank_;
+  int size_;
```

Comment:
What's the reason to remove `const` here?

Reply:
because both has to be set inside the constructor by MPI_comm_rank and MPI_comm_size, but not in the initialization list

Reply:
These can be changed back to const now.